pex/testing.py

```diff
@@ -99,30 +99,47 @@ def write_zipfile(directory, dest, reverse=False):
   return dest
 
 
-PROJECT_CONTENT = {
-  'setup.py': dedent('''
+@contextlib.contextmanager
+def make_project(name='my_project',
+                 version='0.0.0',
+                 zip_safe=True,
+                 install_reqs=None,
+                 extras_require=None):
+
+  project_content = {
+    'setup.py': dedent('''
       from setuptools import setup
 
       setup(
-          name=%(project_name)r,
-          version=%(version)r,
-          zip_safe=%(zip_safe)r,
-          packages=['my_package'],
-          scripts=[
-              'scripts/hello_world',
-              'scripts/shell_script',
-          ],
-          package_data={'my_package': ['package_data/*.dat']},
-          install_requires=%(install_requires)r,
+        name=%(project_name)r,
+        version=%(version)r,
+        zip_safe=%(zip_safe)r,
+        packages=[%(project_name)r],
+        scripts=[
+          'scripts/hello_world',
+          'scripts/shell_script',
+        ],
+        package_data={%(project_name)r: ['package_data/*.dat']},
+        install_requires=%(install_requires)r,
+        extras_require=%(extras_require)r,
       )
-  '''),
-  'scripts/hello_world': '#!/usr/bin/env python\nprint("hello world!")\n',
-  'scripts/shell_script': '#!/usr/bin/env bash\necho hello world\n',
-  'my_package/__init__.py': 0,
-  'my_package/my_module.py': 'def do_something():\n  print("hello world!")\n',
-  'my_package/package_data/resource1.dat': 1000,
-  'my_package/package_data/resource2.dat': 1000,
-}
+    '''),
+    'scripts/hello_world': '#!/usr/bin/env python\nprint("hello world!")\n',
+    'scripts/shell_script': '#!/usr/bin/env bash\necho hello world\n',
+    os.path.join(name, '__init__.py'): 0,
```

Comment:
NB: The point here was just to ensure each project used a unique root package to enable tests that need to check for the presence of a project on sys.path by performing an import. Previously all projects piled on `my_package`.

Reply:
You're wrong, you were right in the 1st place:
https://github.com/pantsbuild/pex/blob/87ea9b5d8d37de92debb5090690c80a623515718/pex/testing.py#L79-L80

Reply:
That would explain my intuition - think I had seen that a month ago and forgotten that's how it works, but was in the back of my mind.